File: solutions/328-odd-even-linked-list/odd-even-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* oddEvenList(struct ListNode* head) {
    if (head == NULL || head->next == NULL) {
        return head;
    }
    struct ListNode* odd = (struct ListNode*)malloc(sizeof(struct ListNode));
    odd->val = 0;
    odd->next = NULL;
    struct ListNode* odd_ptr = odd;
    struct ListNode* even = (struct ListNode*)malloc(sizeof(struct ListNode));
    even->val = 0;
    even->next = NULL;
    struct ListNode* even_ptr = even;
    int idx = 1;
    
    while (head != NULL) {
        if (idx % 2 == 0) {
            even_ptr->next = head;
            even_ptr = even_ptr->next;
        } else {
            odd_ptr->next = head;
            odd_ptr = odd_ptr->next;
        }
        head = head->next;
        idx++;
    }
    even_ptr->next = NULL; 
    odd_ptr->next = even->next;
    struct ListNode* newHead = odd->next; 
    free(odd);
    free(even);
    
    return newHead;
}

File: solutions/328-odd-even-linked-list/odd-even-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode OddEvenList(ListNode head) {
        if(head == null || head.next == null)return head;
        ListNode odd = new ListNode(0);
        ListNode odd_ptr = odd;
        ListNode even = new ListNode(0);
        ListNode even_ptr = even;
        int idx = 1;
        while(head != null){
            if(idx % 2 == 0){
                even_ptr.next = head;
                even_ptr = even_ptr.next;
            }else{
                odd_ptr.next = head;
                odd_ptr = odd_ptr.next;
            }
            head = head.next;
            idx++;
        }
        even_ptr.next = null;
        odd_ptr.next = even.next; 
        return odd.next;
    }
}